File: Leetcode/125.验证回文串.java
class Solution {
    public boolean isPalindrome(String s){
        //原来java里面访问字符串是用charAt的，不能用[]....
        int left = 0, right = s.length() - 1;

        //character.isletterordigit是判断当前的字符是不是数字或字母
        //charactre.tolowercase是把当前的字母转换为小写
        while (left < right){
            while (left < right && !Character.isLetterOrDigit(s.charAt(left))){
                left ++ ;
            }
            while (left < right && !Character.isLetterOrDigit(s.charAt(right))){
                right -- ;
            }
            if (left < right) {
                if (Character.toLowerCase(s.charAt(left)) != Character.toLowerCase(s.charAt(right))){
                    return false;
                }
                left ++ ;
                right -- ;
            }
        }

        return true;
    }
}

File: Leetcode/125.验证回文串.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        //对于字符串的题目，一般有两种方法解决，第一种是预处理，就用一个新的容器对数据进行处理，然后再处理
        //第二种，就是边遍历就边处理，不过最好还是预处理会比较好写一点
        string s1 = "";
        for (char ch:s){
            if (isalnum(ch)){
                s1 += tolower(ch);
            }
        }
        
        int left = 0, right = s1.size() - 1;
        while (left < right){
            if (s1[left] != s1[right]){
                return false;
            }
            left ++ ;
            right -- ;
        }

        return true;
    }
};